presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrderCurveSearcher.java

```diff
@@ -129,6 +112,45 @@ public static ZOrderCurveSearcher createZOrderCurveSearcher(ZOrder zOrder, List<
         return nonNullValues;
     }
 
+    private void combineAddressRanges(List<ZAddressRange<Long>> addressRanges)
+    {
+        if (addressRanges.isEmpty()) {
+            return;
+        }
+
+        addressRanges.sort((range1, range2) -> {
+            if (range1.equals(range2)) {
+                return 0;
+            }
+            if (range1.getMinimumAddress().equals(range2.getMinimumAddress())) {
+                return (range1.getMaximumAddress() > range2.getMaximumAddress()) ? 1 : -1;
+            }
+            return (range1.getMinimumAddress() > range2.getMinimumAddress()) ? 1 : -1;
+        });
+        List<ZAddressRange<Long>> tempRanges = addressRanges.stream().collect(ImmutableList.toImmutableList());
+
+        addressRanges.clear();
+        addressRanges.add(tempRanges.get(0));
+
+        int addressIndex = 0;
+        for (int index = 1; index < tempRanges.size(); index++) {
+            ZAddressRange<Long> prevAddressRange = addressRanges.get(addressIndex);
```

Comment:
Let's use `previousAddressRange` and `currentAddressRange`. We normally don't use abbreviation :)